pkg/etwlogrus/hook.go

```diff
@@ -62,13 +64,29 @@ func (h *Hook) Fire(e *logrus.Entry) error {
 		return nil
 	}
 
+	// Sort the fields by name so they are consistent in each instance
+	// of an event. Otherwise, the fields don't line up in WPA.
+	names := make([]string, 0, len(e.Data))
+	var errv interface{}
+	for k := range e.Data {
+		names := make([]string, 0, len(e.Data))
+		if k == logrus.ErrorKey {
+			// Always put the error last because it is optional in some events.
+			errv = e.Data[k]
+		} else {
+			names = append(names, k)
+		}
+	}
+	sort.Strings(names)
+
 	// Reserve extra space for the message field.
 	fields := make([]etw.FieldOpt, 0, len(e.Data)+1)
-
 	fields = append(fields, etw.StringField("Message", e.Message))
-
-	for k, v := range e.Data {
-		fields = append(fields, etw.SmartField(k, v))
+	for _, k := range names {
+		fields = append(fields, etw.SmartField(k, e.Data[k]))
+	}
+	if errv != nil {
```

Comment:
Not sure we care but if this ever had `data[logrus.ErrorKey] = nil` we would drop it here. Do we want "Error" to be written based on its keys presence or based on the value?

Reply:
Hmm. I guess for consistency I should make it based on presence.